packages/wallet/src/account.ts

```diff
@@ -306,33 +306,24 @@ export class Account extends Signer {
     const authWallet = this.authWallet().wallet
     const authContract = new Contract(authWallet.context.sequenceUtils, walletContracts.sequenceUtils.abi, authWallet.provider)
 
-    let event: any
     if (currentImplementation === wallet.context.mainModuleUpgradable) {
-      // Wallet is deployed on chain, test if given config is the updated one
       if (imageHash(authWallet.config) === currentImageHash[0]) {
         return { ...authWallet.config, address, chainId }
       }
-
-      // The wallet has been updated. Lookup configuration using imageHash from the authChain
-      // which will be the last published entry
-      const filter = authContract.filters.RequiredConfig(null, currentImageHash)
-      const lastLog = await findLatestLog(authWallet.provider, { ...filter, fromBlock: 0, toBlock: 'latest'})
-      if (lastLog === undefined) return undefined
-      event = authContract.interface.decodeEventLog('RequiredConfig', lastLog.data, lastLog.topics)
-    
     } else {
-      // Wallet is undeployed, test if given config is counter-factual config
-      if (addressOf(authWallet.config, authWallet.context).toLowerCase() === address.toLowerCase()) {
+      if (addressOf({ ...authWallet.config, address: undefined }, authWallet.context).toLowerCase() === address.toLowerCase()) {
         return { ...authWallet.config, chainId }
       }
-
-      // The wallet it's using the counter-factual configuration, get the init config from the authChain
-      const filter = authContract.filters.RequiredConfig(address)
-      const lastLog = await findLatestLog(authWallet.provider, { ...filter, fromBlock: 0, toBlock: 'latest'})
-      if (lastLog === undefined) return undefined
-      event = authContract.interface.decodeEventLog('RequiredConfig', lastLog.data, lastLog.topics)
     }
 
+
+    // Get last known configuration
+    const logBlockHeight = (await authContract.lastWalletUpdate(this.address)).toNumber()
+    const filter = authContract.filters.RequiredConfig(this.address)
+    const lastLog = await findLatestLog(authWallet.provider, { ...filter, fromBlock: logBlockHeight, toBlock: logBlockHeight !== 0 ? logBlockHeight : 'latest'})
+    if (lastLog === undefined) { console.log("LOG NOT FOUND"); return undefined }
```

Comment:
lets update this console log to.. `console.warn("publishConfig: wallet config last log not found")`